database/migrations/2020_05_01_120000_create_orders_table.php

```diff
@@ -14,10 +14,25 @@ public function up()
             $table->id();
             $table->string('order_number')->index()->unique(); // Generated by system. This is identifier used to communicate with customers about their order.
             $table->foreignIdFor(app('customer'));
-            $table->unsignedInteger('amount'); // Amount is in cents. It is net, excluding taxes and fees. An accessor for total_amount adds the 3 columns
-            $table->unsignedInteger('total_taxes'); // Taxes are in cents.
-            $table->unsignedInteger('total_fees'); // Processing Fees are in cents.
-            $table->unsignedInteger('partial_redemption_voucher_id')->nullable();
+
+            // TODO - TBD - how to handle linkage for shipping address
+            // TODO - TBD - how to integrate w/statuses
+
+            $table->unsignedInteger('shipping')->default(0);
+            $table->unsignedInteger('shipping_discounts')->default(0);
+
+            // Additional order-level discounts not reflected in order item discounts
+            $table->unsignedInteger('order_discounts')->default(0);
+
+            // Direct calculation from sum(amount)/sum(amount_discounts) in order_items
+            // TODO - TBD - duplicate values for easy reporting?  Or always calculate?
+            $table->unsignedInteger('item_amount')->default(0);
+            $table->unsignedInteger('item_amount_discounts')->default(0);
```

Comment:
This is a TBD for cart/order - there are a few values in that are completely derived from cart item / order item values.  Do you like the duplication in the summary cart / order record?   So long as direct DB manipulations are avoided, it should be easy to ensure the duplication is always correct.

Reply:
@drewroberts  - any thoughts on this? should we keep the duplicated values in the parent cart/order?

Reply:
FWIW - I ended up keeping the duplicate values w/automatic updating.